assignments/week_4/week_4_exercises/exercise_4_1/exercise_4_1.md

```diff
@@ -0,0 +1,127 @@
+```c++
+void callback_function(std_msgs::String string_msg){
+    ROS_INFO_STREAM(string_msg.data);
+}
+```
+
+# Creating the subscriber object
+To create a subscriber, you will use the `subscribe` method of the `NodeHandle` object
+already created for you. The type of the object created will to be `ros::Subscriber`.
+```c++
+ros::Subscriber subscriber = p_nh.subscribe(...);
+```
+
+The arguments to this function are:
+* The topic we subscribe to, which for this exercise is `"/string_topic"`
+* The queue size we want, which should be 1
+* A reference to the callback function the subscriber should use. In C++, the way
+you reference a function is by using the `&` character right before the name of the 
+function, WITHOUT the parenthesis, otherwise the function will be called.
+
+In the end, your subscriber construction should look like this:
+```c++
+ros::Subscriber subscriber = p_nh.subscribe("/string_topic", 1, &callback_function);
+```
+
+# Compile the node and package
+To compile the `week_4_exercises` package, we need to create a workspace. 
+To make and compile the workspace, do the following:
+* create a workspace folder somewhere, named `training_ws`
+* create a folder inside `training_ws` named `src`
+* make a copy `week_4_exercises` and put it in `training_ws/src`
+* open a terminal and change directory to `training_ws`
+* run `catkin_make`
+
+The package should be compiled! If you run into problems, be sure to double check all your changes.
+
+# Run the node using roslaunch
+Now that we've written the node, we need to test it. To do this we'll use roslaunch.
+First, since we've built in a new workspace, we need to tell ros where our files are.
+To do this, run this command in a terminal inside `training_ws`
+```shell script
+$ source devel/setup.bash
+```
+
+This lets the terminal you ran the command in find and launch the ros nodes in `training_ws`. 
+To launch the node, first you need to run `roscore` in a separate terminal. In Terminator, use
+`ctrl+shift+e` to open another terminal instance. 
+
+In one terminal, run 
+```shell script
+$ roscore
+```
+
+In the other, run
+```shell script
+$ roslaunch week_4_exercises exercise_4_1.launch
+```
+
+If you did everything correctly, you should see the following output:
+```
+[ INFO] [...]: EXERCISE: 1
+[ INFO] [...]: wrestling
+[ INFO] [...]: navigation
+[ INFO] [...]: soccer
+[ INFO] [...]: racecar
+[ INFO] [...]: computer
+[ INFO] [...]: bee
+[ INFO] [...]: 
+[ INFO] [...]: testing testing 123
+```
+
+Use `ctrl+c` to stop the nodes and `roscore`.
+
+# Find the log file created when the node was running
+To verify that you completed this assignment, you will need to also turn in the log file
+generated from running the node. To find the most recent log file, go to `~/.ros/log/latest`.
+There will be several log files, but the one we want is `rosout.log`. Open it and verify that it 
+contains the messages you saw in the console.
+
+# Turn in the node and log file to show your node worked
```

Comment:
tell them what command to use explicitly

Reply:
Don't know exactly what you mean here.

Reply:
.ros is a hidden folder so it showing up in file explorer is non trivial. I mean literally write out the cp command.